service/validator.go

```diff
@@ -113,9 +129,63 @@ func (sv ValidatorConfig) ValidateRequestAndReturnToken(r *http.Request) (TokenW
 		token: token,
 	}
 
+	err = sv.validateAudiences(*tokenWithClaims, sv.audience.All())
+	if err != nil {
+		return nil, err
+	}
+
 	return tokenWithClaims, nil
 }
 
+// ValidateAudiences ...
+func (sv ValidatorConfig) validateAudiences(tokenWithClaims tokenWithClaims, audiences []string) error {
+	payload, err := tokenWithClaims.Payload()
+	if err != nil {
+		return err
+	}
+
+	var audiencesInToken []string
+
+	switch v := payload["aud"].(type) {
+	default:
+		fmt.Printf("unexpected type %T", v)
+	case string:
+		audience, ok := payload["aud"].(string)
+		if !ok {
+			panic("type assertion failed")
+		}
+		audiencesInToken = []string{audience}
+	case []interface{}:
+		audienceInterface, ok := payload["aud"].([]interface{})
+		if !ok {
+			panic("type assertion failed")
+		}
+		audiencesInToken = make([]string, len(audienceInterface))
+		for i, v := range audienceInterface {
+			audience, ok := v.(string)
+			if !ok {
+				panic("type assertion failed")
+			}
+			audiencesInToken[i] = audience
+		}
+	}
+
+	if len(audiencesInToken) > 0 {
```

Comment:
We need to validate the audiences in the config regardless whether the token has any. If we expect 2 audiences, but the token has none, it should fail